src/notation/internal/notationselectionrange.cpp

```diff
@@ -221,7 +221,7 @@ const
         mu::engraving::System* currentSegmentSystem = segment->measure()->system();
 
         mu::engraving::Segment* nextSegment = segment->next1MMenabled();
-        mu::engraving::System* nextSegmentSystem = nextSegment->measure()->system();
+        mu::engraving::System* nextSegmentSystem = nextSegment ? nextSegment->measure()->system() : segment->measure()->system();
```

Comment:
Why is it null?

Reply:
I just noticed that Segment::next1MM() (which is called by Segment::next1MMenabled()) can explicitely return a null, so its output should be null-checked. Why exactly it returns null in this case, I'm not sure, I can look into it

Reply:
>Why exactly it returns null in this case, I'm not sure

We should find out why it returns null

Reply:
It seems that we were skipping the exit condition of the loop, so it was running until the end of the score (where a null return is normal). Now it should be good

Reply:
1. You should create a temp variable for this expression outside the loop
`rangeEndSegment->tick()`

2. How is this change related to Ctrl+Z? I don't get it from the commit message

Also, please notice that there is some mess in the commit message

<img width="321" alt="Screen Shot 2022-11-22 at 11 35 46" src="https://user-images.githubusercontent.com/67867444/203279071-a9f9bd79-b2a4-42dc-b70a-e5033545074e.png">

Reply:
I think I was finally able to find the root problem: when doing ctrl+Z we were tying to reset the selection, but we were computing the wrong end-segment in `Selection::updateSelectedElements()`, which was messing up the rest. Now it's fixed (but I still changed the exit condition of the loop to make it safer).